test/diff_test.go

```diff
@@ -381,7 +381,7 @@ func TestDiffWithEmptyDirectories(t *testing.T) {
 		"e/f": "f2",
 	}, []string{"b/c", "b/d"})
 
-	d2, s2, err := diff.Diff(diff.WalkChan(tmpDir, []string{}), diff.SummaryChan(s1))
+	d2, s2, err := diff.Diff(diff.WalkChan(tmpDir, []string{}, s1.LatestModTime), diff.SummaryChan(s1))
 	if err != nil {
 		t.Fatalf("failed to run diff: %v", err)
 	}
```

Comment:
Can we contrive any tests that stress the timestamp comparison logic to capture some of the edge cases mentioned in the git design doc?

Reply:
The tests - at least locally - already stress the `>=` clause by landing with the exact same nanosecond on some runs.

I'll add a couple new tests with different mod times passed here to properly capture how files are skipped with old mod times.